### PR TITLE
feat(blockage_diag): time series blockage diag

### PR DESCRIPTION
**Pull request by swiftfile**
## Description
This change in blockage-diag is to reduce false positives.
In the previous implementation, areas that did not return Lidar reflections, such as the sky, were also considered as blockages (false positives). because they were considered as blockages where there were no reflections in a single frame.
However, the new implementation detects blockages from multi-frame depth maps, so even areas like the sky are less likely to be considered blockage.
This is because even if a single frame points to the sky and has no reflections, another frame will have reflections from a building or traffic sign.

## Related links

[Pull Request #63 · tier4/aip_launcher - feat: add parameter about blockage_diag by swiftfile](https://github.com/tier4/aip_launcher/pull/63)
#1002  (this PRis invalid. So,I close this PR)
## Tests performed
time_series_blockage
The top row is the depth map, the middle row is single frame blockage detection (old implementation), and the bottom row is multiple frame blockage detection (new implementation).
Take the right column as an example.
When the lidar is pointing to the sky all the time and no reflection is obtained, both single-frame and multiple-frame blockage detection are judged to be blockage (red pixel). However, in multi-frame blockage detection, if there is a reflection from a building or other object in another frame, it will not be detected as a blockage (blue pixel).
In addition, this implementation will not recognize a blockage as not blockage, even when there is actually a blockage, such as in the left column.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: sensing/pointcloud_preprocessor/src/blockage_diag/blockage_diag_nodelet.cpp [hunk 1]
```diff
@@ -163,6 +167,32 @@ void BlockageDiagComponent::filter(
...
+      }
+      for (const auto & binary_mask : no_return_mask_buffer) {
+        time_series_blockage_mask += binary_mask;
+      }
+      cv::inRange(
+        time_series_blockage_mask, no_return_mask_buffer.size() - 1, no_return_mask_buffer.size(),
+        no_return_mask_result);
+    } else {
+      no_return_mask.copyTo(no_return_mask_result);
+    }
```
[the remark this answers is not in the document]
Comment (swiftfile) [reply sensing/pointcloud_preprocessor/src/blockage_diag/blockage_diag_nodelet.cpp:187]: For now, I have reflected your comment.
https://github.com/autowarefoundation/autoware.universe/pull/1004/commits/52c78bb5af8693e38d545e1c7c52ef4a01870e88

[the remark this answers is not in the document]
Comment (badai-nguyen) [reply sensing/pointcloud_preprocessor/src/blockage_diag/blockage_diag_nodelet.cpp:187]: LGTM 
Do you have other comments? @miursh @drwnz

**Comment by badai-nguyen** [comment]
LGTM 
Do you have other comments? @miursh @drwnz

**Comment by miursh** [comment]
Please give me some more days :pray:  Sorry for my late response

**Review by miursh — APPROVED** [review]
LGTM. If @badai-nguyen approves, please merge this PR.

**Review by badai-nguyen — APPROVED** [review]

**Review by badai-nguyen — APPROVED** [review]

**Review by badai-nguyen — APPROVED** [review]

**Review by badai-nguyen — APPROVED** [review]

**Pull request merged**